src/metrics.ts

```diff
@@ -63,8 +65,260 @@ function initializeMetricMaps() {
   };
 }
 
-// take a start and end board position and return the distance moveds
-function getMoveDistance(start: string, end: string) {}
+// take a start and end board position and return the distances moved
+export async function getMoveDistance(filePath: string) {
+  for await (const game of gameChunks(filePath)) {
+    try {
+      const chess = new Chess(); // Create a new instance of the Chess class
+      for (const line of game) {
+        // Ignore header information
+        if (!line.startsWith('[') && line.trim() !== '') {
+          const moves = line.split(' ').filter(move => !move.endsWith('.') && !['1-0', '0-1', '1/2-1/2'].includes(move));
+          for (const move of moves) {
+            //console.log(`Processing move: ${move}`);
+            chess.move(move);
+          }
+        }
+      }
+
+      const ingestedMoves = chess.history({ verbose: true })
+      //console.log(ingestedMoves)
+      
+      const pieces: { [key: string]: { initialPosition: string; currentPosition: string; distanceMoved: number } } = {
```

Comment:
this is good to start, how do you want to handle promotions?

Reply:
haven't gotten there yet